src/lib/Microsoft.Health.Extensions.Fhir.R4/BearerTokenAuthorizationMessageHandler.cs

```diff
@@ -0,0 +1,54 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using EnsureThat;
+using Microsoft.Health.Common.Telemetry;
+using Microsoft.Health.Extensions.Fhir.Telemetry.Metrics;
+using Microsoft.Health.Logging.Telemetry;
+
+namespace Microsoft.Health.Extensions.Fhir
+{
+    public class BearerTokenAuthorizationMessageHandler : DelegatingHandler
+    {
+        public BearerTokenAuthorizationMessageHandler(Uri uri, TokenCredential tokenCredentialProvider, ITelemetryLogger logger)
+        {
+            TokenCredential = EnsureArg.IsNotNull(tokenCredentialProvider, nameof(tokenCredentialProvider));
+            Uri = EnsureArg.IsNotNull(uri);
+            Scopes = new string[] { Uri + ".default" };
```

Comment:
No, in my testing the string concatenation was not adding a forward slash to the URI if it wasn't provided. I've updated the code to append a forward slash when the URI doesn't have one. 

I chose to go with the .EndsWith() method for strings instead of System.TryCreateUri because Scopes is a string array

Reply:
I agree with Rob, using Uri.TryCreate should be cleaner and safer.

Example 
var baseUri = new Uri("[https://test.com](https://test.com)");
Uri.TryCreate(baseUri, ".default", out Uri newUri);

or in this code it would be 
Uri.TryCreate(Uri, ".default", out Uri scopeUri);
Scopes = new string [] { scopeUri.ToString() };